### PR TITLE
More Method -> AssocFn renaming

### PR DESCRIPTION
**Pull request by mark-i-m**
r? @Centril @eddyb 

cc #60163 

Blocked on #69674

### CODE REVIEW

File: src/librustc_hir/hir.rs [hunk 1]
```diff
@@ -1898,7 +1898,7 @@ pub enum ImplItemKind<'hir> {
     /// of the expression.
     Const(&'hir Ty<'hir>, BodyId),
     /// A method implementation with the given signature and body.
```
Comment (eddyb) [inline src/librustc_hir/hir.rs]: The comment here is not changed, even if the one for `TraitItemKind` is (I think this is from the other PR, actually).

Comment (Centril) [reply src/librustc_hir/hir.rs]: r=me with the comment changed.

**Comment by mark-i-m** [comment]
@rustbot modify labels: +S-blocked -S-waiting-on-author

@Centril Done! But we should still wait for the other PR, since this is rebased on that one.

File: src/librustc_metadata/rmeta/mod.rs [hunk 2]
```diff
@@ -306,7 +306,7 @@ enum EntryKind {
     Generator(hir::GeneratorKind),
     Trait(Lazy<TraitData>),
     Impl(Lazy<ImplData>),
-    Method(Lazy<MethodData>),
+    AssocFn(Lazy<MethodData>),
```
Comment (eddyb) [inline src/librustc_metadata/rmeta/mod.rs]: `MethodData` should be `AssocFnData`.

**Review by eddyb — APPROVED** [review]

**Comment by eddyb** [comment]
r=me after fixing https://github.com/rust-lang/rust/pull/69738#discussion_r392661817

**Review by eddyb — APPROVED** [review]

**Pull request merged**